src/Compilers/CSharp/Portable/Binder/LockBinder.cs

```diff
@@ -55,9 +57,92 @@ internal override BoundStatement BindLockStatementParts(BindingDiagnosticBag dia
                 hasErrors = true;
             }
 
+            if (exprType?.IsWellKnownTypeLock() == true &&
+                CheckFeatureAvailability(exprSyntax, MessageID.IDS_NativeLock, diagnostics) &&
```

Comment:
>CheckFeatureAvailability(exprSyntax, MessageID.IDS_NativeLock, diagnostics)  [](http://example.com/codeflow?start=16&length=76)

I do not think the language check should be part of the condition. We shouldn't change how we bind things, we should simply report an error if an old language version is targeted.